test/m1_facilitator/repositories/WithdrawIntentRepository/util.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+
+import assert from '../../../test_utils/assert';
+import WithdrawIntent from '../../../../src/m1_facilitator/models/WithdrawIntent';
+
+const Util = {
+  assertWithdrawIntentAttributes(
+    inputWithdrawIntent: WithdrawIntent,
+    expectedWithdrawIntent: WithdrawIntent,
+  ): void {
+    assert.strictEqual(
+      inputWithdrawIntent.intentHash,
+      expectedWithdrawIntent.intentHash,
+      'Mismatch in intent hash.',
+    );
+
+    assert.strictEqual(
+      inputWithdrawIntent.messageHash,
+      expectedWithdrawIntent.messageHash,
+      'Mismatch in message hash.',
+    );
+
+    assert.strictEqual(
+      inputWithdrawIntent.tokenAddress,
+      expectedWithdrawIntent.tokenAddress,
+      'Mismatch in token address.',
+    );
+
+    if (inputWithdrawIntent.amount && expectedWithdrawIntent.amount) {
```

Comment:
Why is this if condition needed?

Reply:
It is needed to check the whether the expected withdraw amount and input withdraw amount both are not null. If both are not null, then only it will go for assert further.

Reply:
This is incorrect, please remove the if condition.